### PR TITLE
convert very long messages to payloads

### PR DESCRIPTION
**Pull request by 2002Bishwajeet**

### CODE REVIEW

**Comment by 2002Bishwajeet** [comment]
Needs some help here. Looks like the cache gets disposed on every refresh and thus the message gets undefined which leads to hard crash. Can you help me out here please. Attached ss here
image

**Comment by stef-coenen** [comment]
> Needs some help here. Looks like the cache gets disposed on every refresh and thus the message gets undefined which leads to hard crash. Can you help me out here please. Attached ss here

Definitely not a disposed cache problem. But there was a missing `dsrToMessage` call when getting the cache up to date. Which wasn't a problem and was more efficient. But of course doesn't cover messages with a default payload

File: packages/mobile/src/provider/chat/ChatProvider.ts
Comment (stef-coenen) [inline packages/mobile/src/provider/chat/ChatProvider.ts:298]: Maybe we ought to do it a bit differently for chat. I was thinking to keep the parts that fit still in the content; And add a hard cut off on the message; That way we can then extend the message renderers to only fetch the full payload when expanded by a user;

And that way we don't need to fetch the payloads of every message when they're never rendered (older page, or conversation isn't opened soon anymore);

**Review by stef-coenen — CHANGES_REQUESTED** [review]

File: packages/mobile/src/provider/chat/ChatProvider.ts
Comment (2002Bishwajeet) [reply packages/mobile/src/provider/chat/ChatProvider.ts:298]: I initially gave a similar thought about this to keep it that way and fetch the payload when we click on the expand button but I kept it similar to the way we do for posts at the moment. 

Maybe we refactor both the same way?

Comment (stef-coenen) [reply packages/mobile/src/provider/chat/ChatProvider.ts:298]: Don't see a reason in refactoring those together. And for posts that will be much harder to do, as splitting up Rich Text articles is a lot more elaborate vs plain text.

So for posts I would prefer to wait till we got a final decision from the BE on the max size of a header.

Comment (2002Bishwajeet) [reply packages/mobile/src/provider/chat/ChatProvider.ts:298]: makes sense

File: packages/mobile/src/components/Chat/ChatDetail.tsx
Comment (stef-coenen) [inline packages/mobile/src/components/Chat/ChatDetail.tsx:909]: Editing a property from within the component is a big no-no I'm afraid. Why don't you pass the `currentMessage` prop into a state variable `message` so you can do a nice `setMessage` when you've fetched the extra payload

File: packages/mobile/src/hooks/chat/useChatMessage.ts
Comment (stef-coenen) [inline packages/mobile/src/hooks/chat/useChatMessage.ts:467]: Perhaps move this part to a separate hook? It uses different props to fetch so only makes the hook more complex than it actually is

**Review by stef-coenen — CHANGES_REQUESTED** [review]

**Comment by stef-coenen** [comment]
You'll need to change the `dsrToMessage` within ChatProvider as well as that will already fetch the default payload

**Comment by 2002Bishwajeet** [comment]
> You'll need to change the `dsrToMessage` within ChatProvider as well as that will already fetch the default payload

But we use a different `payloadKey` for expanded chat messages, so no changes needed right?

**Comment by stef-coenen** [comment]
> But we use a different payloadKey for expanded chat messages, so no changes needed right?
True, but we can't have both right? Why not store the expanded text in the default payload key? That way we can be explicit about it, and that way we don't have to align the releases of web and mobile.

File: packages/mobile/src/provider/chat/ChatProvider.ts
Comment (stef-coenen) [inline packages/mobile/src/provider/chat/ChatProvider.ts:98]: You can remove this one now right?

File: packages/mobile/src/components/Chat/ChatDetail.tsx
Comment (stef-coenen) [inline packages/mobile/src/components/Chat/ChatDetail.tsx:862]: This should now be the DEFAULT_PAYLOAD_KEY right?

File: packages/mobile/src/provider/chat/ChatProvider.ts
Comment (stef-coenen) [inline packages/mobile/src/provider/chat/ChatProvider.ts:310]: This should now be the DEFAULT_PAYLOAD_KEY right?

File: packages/mobile/src/components/Chat/ChatDetail.tsx
Comment (stef-coenen) [inline packages/mobile/src/components/Chat/ChatDetail.tsx:866]: This should now be the DEFAULT_PAYLOAD_KEY right?

**Review by stef-coenen — CHANGES_REQUESTED** [review]

**Review by stef-coenen — APPROVED** [review]

**Pull request merged**
